cache_store.js

```diff
@@ -0,0 +1,60 @@
+module.exports = CacheStore;
+
+function CacheStore(store) {
+    this.store = store;
+    // Only stores sets that have been loaded
+    // Contains map of key -> bool, where bool isn't important
+    this.cache = {};
```

Comment:
Since keys can be arbitrary strings, consider using `Object.create(null)` here (and `Object.prototype.hasOwnProperty` in the functions below)